In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('/content/iris_dataset_1.csv')

In [3]:
df.head()

,sepal_length,sepal_width,petal_length,petal_width,species
0,5.1,3.5,1.4,0.2,setosa
1,4.9,3.0,1.4,0.2,setosa
2,4.7,3.2,1.3,0.2,setosa
3,4.6,3.1,1.5,0.2,setosa
4,5.0,3.6,1.4,0.2,setosa


In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal_length  150 non-null    float64
 1   sepal_width   150 non-null    float64
 2   petal_length  150 non-null    float64
 3   petal_width   150 non-null    float64
 4   species       150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


In [5]:
cols = ['sepal_length', 'sepal_width', 'petal_length', 'petal_width']
df[cols].describe()

,sepal_length,sepal_width,petal_length,petal_width
count,150.000000,150.000000,150.000000,150.000000
mean,5.843333,3.057333,3.758000,1.199333
std,0.828066,0.435866,1.765298,0.762238
min,4.300000,2.000000,1.000000,0.100000
25%,5.100000,2.800000,1.600000,0.300000
50%,5.800000,3.000000,4.350000,1.300000
75%,6.400000,3.300000,5.100000,1.800000
max,7.900000,4.400000,6.900000,2.500000


In [6]:
df[cols].mode().iloc[0]

,0
sepal_length,5.0
sepal_width,3.0
petal_length,1.4
petal_width,0.2


In [7]:
df.groupby('species')[cols].mean()

,sepal_length,sepal_width,petal_length,petal_width
species,,,,
setosa,5.006,3.428,1.462,0.246
versicolor,5.936,2.770,4.260,1.326
virginica,6.588,2.974,5.552,2.026


**Summary of the dataset**

I calculated descriptive statistics for the 4 numeric columns in the Iris dataset — sepal_length, sepal_width, petal_length, and petal_width.

-For sepal_length and sepal_width, the mean, median, and mode values are all quite close to each other. This shows that their distribution is roughly symmetric, without any major skew in these two columns. The standard deviation is also low (0.83 and 0.44), meaning the values stay fairly close to their average.

-For petal_length and petal_width, I noticed a bigger difference between the mean, median, and mode. For example, petal_length had a mean of 3.76 but a mode of only 1.4 — at first this looked like the data was skewed. But when I grouped the data by species using groupby('species'), it became clear that there are actually 3 different species with very different petal sizes:

-Setosa has the smallest petals (length ~1.46, width ~0.25)
Versicolor is medium-sized (length ~4.26, width ~1.33)
Virginica has the largest petals (length ~5.55, width ~2.03)

**-Looking at the 25th and 75th percentiles also confirms this — the middle 50% of petal_length values (1.6 to 5.1) is a much wider range compared to sepal_width (2.8 to 3.3), which shows petal measurements are far more spread out across the different species.**

-So the gap wasn't due to skewness in a single distribution — it was because data from three different groups was mixed together.

My takeaway: If mean, median, and mode differ a lot for a column, it's not always correct to assume the data is skewed. It's important to first check whether there's a hidden category or group in the data causing that pattern. In the Iris dataset, species turned out to be exactly that hidden factor.